app/data_definitions/base.rb

```diff
@@ -27,12 +44,33 @@ def get_child(field)
       node
     end
 
-    # rubocop:disable Rails/OutputSafety
-    def get_child_value(field, unescaped=false)
-      node = get_child(field)
-      value = node.content
-      unescaped ? value.html_safe : value
+    def select_single_node_value(xpath)
+      # Mirrors Cascade Velocity helper: $_XPathTool.selectSingleNode(node, xpath).value
+      # Returns string value.
+      xpath = format('//%s', xpath) unless xpath.start_with?('/')
+      node = @data_document.at_xpath(xpath)
+      node.nil? ? '' : node.content
+    end
+
+    def select_nodes(xpath)
+      # Mirrors Cascade Velocity helper: $_XPathTool.selectNodes(node, xpath).
+      # Returns DataDefinitions:NodeSet array.
+      xpath = format('//%s', xpath) unless xpath.start_with?('/')
+      @data_document.xpath(xpath)
+    end
+  end
+
+  # Extend Nokogiri Nodeset
+  module NokogiriExtensions
+    module Element
+      def select_single_node_value(xpath)
+        node = at_xpath(xpath)
+        raise "Node not found for xpath #{xpath}." unless node
+        node.content
+      end
     end
-    # rubocop:enable Rails/OutputSafety
   end
+
+  # Monkey-patch it now!
+  Nokogiri::XML::Element.include DataDefinitions::NokogiriExtensions::Element
```

Comment:
Not saying anything is wrong, I just see a comment saying that something is monkey patched. What's the reasoning?